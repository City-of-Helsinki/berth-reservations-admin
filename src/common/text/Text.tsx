import React from 'react';
import classNames from 'classnames';

import styles from './text.module.scss';

export interface TextProps {
  children: React.ReactNode;
  className?: string;
  color?: 'standard' | 'brand' | 'critical' | 'secondary' | 'info' | 'gray';
  as?: 'span' | 'em' | 'strong' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  size?: 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'xxl' | 'xxxl';
  weight?: 'bold' | 'normalWeight' | 'light';
  italic?: boolean;
  underlined?: boolean;
  uppercase?: boolean;
}

const Text = ({
  className,
  color = 'standard',
  size,
  as: Element = 'span',
  weight,
  italic,
  underlined,
  uppercase,
  children,
}: TextProps) => {
  return (
    <Element
      className={classNames(
        styles.text,
        styles[color],
        styles[Element],
        size && styles[size],
        weight && styles[weight],
        {
          [styles.italic]: italic,
          [styles.underlined]: underlined,
          [styles.uppercase]: uppercase,
        },
        className
      )}
    >
      {children}
    </Element>
  );
};

export default Text;
