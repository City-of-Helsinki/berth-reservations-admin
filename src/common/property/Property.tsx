import React from 'react';
import classNames from 'classnames';

import Text from '../text/Text';
import styles from './property.module.scss';
import IconWrapper from '../iconWrapper/IconWrapper';
import { IconProps } from '../icons';

export interface PropertyProps {
  active?: boolean;
  icon?: (props: IconProps) => React.ReactElement | null;
  counter?: number | null;
  label: string;
  className?: string;
}

const Property = ({ active = true, counter = 0, label, icon, className }: PropertyProps) => {
  return (
    <div className={classNames(styles.property, className)}>
      {icon ? (
        <IconWrapper color={active ? 'standard' : 'secondary'} icon={icon} outlined size="m" />
      ) : (
        <Text color={active ? 'standard' : 'secondary'} as="strong" size="xxxl">
          {counter}
        </Text>
      )}

      <Text className={styles.label} weight="bold" size="s" color={active ? 'standard' : 'secondary'}>
        {label}
      </Text>
    </div>
  );
};

export default Property;
