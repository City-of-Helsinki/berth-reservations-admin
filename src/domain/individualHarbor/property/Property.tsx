import React from 'react';
import classNames from 'classnames';

import Text from '../../../common/text/Text';
import styles from './property.module.scss';

export interface IconProps {
  disabled?: boolean;
  outlined?: boolean;
  width?: string;
  height?: string;
  size?: 'small' | 'standard' | 'large';
  color?: 'standard' | 'brand' | 'critical' | 'secondary' | 'info';
}

const IconOutline: React.SFC<IconProps> = ({ children, disabled }) => {
  return (
    <div
      className={classNames(styles.icon, styles.outlined, {
        [styles.disabled]: disabled,
      })}
    >
      {children}
    </div>
  );
};

interface PropertyProps {
  active?: boolean;
  Icon?: Function;
  counter?: number;
  label: string;
}

const Property: React.SFC<PropertyProps> = ({
  active = true,
  counter = 0,
  label,
  Icon,
}) => {
  return (
    <div className={styles.property}>
      {Icon ? (
        <IconOutline>
          <Icon
            className={classNames(styles.icon, styles.outline, {
              [styles.disabled]: false,
            })}
          />
        </IconOutline>
      ) : (
          <Text as="h3" size="xxxl" color={active ? 'standard' : 'secondary'}>
            {counter}
          </Text>
        )}

      <Text as="strong" size="s" color={active ? 'standard' : 'secondary'}>
        {label}
      </Text>
    </div>
  );
};

export default Property;
