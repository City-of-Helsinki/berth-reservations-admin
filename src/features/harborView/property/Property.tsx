import React from 'react';
import classNames from 'classnames';

import Text from '../../../common/text/Text';
import styles from './property.module.scss';
import Icon, { IconShapes } from '../../../common/icons/Icon';

interface PropertyProps {
  active?: boolean;
  iconShape?: IconShapes;
  counter?: number;
  label: string;
  className?: string;
}

const Property: React.SFC<PropertyProps> = ({ active = true, counter = 0, label, iconShape, className }) => {
  return (
    <div className={classNames(styles.property, className)}>
      {iconShape ? (
        <Icon color={active ? 'standard' : 'secondary'} shape={iconShape} outlined size="large" />
      ) : (
        <Text color={active ? 'standard' : 'secondary'} as="strong" size="xxxl">
          {counter}
        </Text>
      )}

      <Text className={styles.label} weight="bold" size="s" color={active ? 'standard' : 'secondary'}>
        {label}
      </Text>
    </div>
  );
};

export default Property;
