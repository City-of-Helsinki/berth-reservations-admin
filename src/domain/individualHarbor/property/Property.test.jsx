import React from 'react';
import { shallow } from 'enzyme';

import Property from './Property';
import Text from '../../../common/text/Text';

describe('Property', () => {
  const getWrapper = (props = { iconName: 'fence', label: 'Test' }) =>
    shallow(<Property {...props} />);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders Text component with strong value for as prop when iconName is not supplied', () => {
    const counter = 9;
    const wrapper = getWrapper({ counter });

    expect(
      wrapper
        .find(Text)
        .first()
        .prop('as')
    ).toBe('strong');
  });
});
