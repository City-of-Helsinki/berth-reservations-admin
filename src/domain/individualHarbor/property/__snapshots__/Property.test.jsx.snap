// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property renders normally 1`] = `"<div class=\\"property\\"><div class=\\"icon large standard\\"><svg class=\\"shape\\">fence.svg</svg></div><strong class=\\"text standard strong s\\">Test</strong></div>"`;
